src/main/java/org/jenkinsci/test/acceptance/po/Jenkins.java

```diff
@@ -112,7 +113,14 @@ public void restart() {
         visit("restart");
         clickButton("Yes");
 
-        waitFor(driver, not(hasContent("Please wait")), JenkinsController.STARTUP_TIMEOUT);
+        try {
+            waitFor(driver, not(hasContent("Please wait")), JenkinsController.STARTUP_TIMEOUT);
+        }catch(TimeoutException e) {
+            //Let's try to avoid false negatives or not auto refresh
+            visit(driver.getCurrentUrl());
+            //we wait 10 seconds for refresh things.
+            waitFor(driver, hasContent("New Item"), 10);
```

Comment:
@lordofthejars, you have missed my point. We are trying to detect when "Please wait" turns into the real page, but after your fix there are 2 mechanisms to detect that: `not(hasContent("Please wait"))` and `hasContent("New Item")`. We should agree on one. Anyway, there seems to be a better alternative, see my PR comment bellow.

And as James pointed out there is a way to save some time in happy scenario using `waitFor().withTimeout(..., ...).until(...)`.